Fitness/Controllers/HomeViewController+Extensions.swift

```diff
@@ -0,0 +1,224 @@
+
+import UIKit
+
+extension HomeViewController: UICollectionViewDataSource {
+
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return sections.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let sectionType = sections[indexPath.section]
+        switch sectionType {
+        case .checkIns:
+            // swiftlint:disable:next force_cast
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.checkInsListCellIdentifier, for: indexPath) as! CheckInsListCell
+            cell.configure(for: habits)
+            return cell
+        case .allGyms:
+            // swiftlint:disable:next force_cast
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.gymsListCellIdentifier, for: indexPath) as! GymsListCell
+            cell.delegate = self
+            cell.configure(for: self.favoriteGyms)
+            return cell
+        case .todaysClasses:
+            // swiftlint:disable:next force_cast
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.todaysClassesListCellIdentifier, for: indexPath) as! TodaysClassesListCell
+            cell.delegate = self
+            cell.configure(for: gymClassInstances)
+            return cell
+        default:
+            // swiftlint:disable:next force_cast
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.prosListCellIdentifier, for: indexPath) as! ProsListCell
+            cell.delegate = self
+            cell.configure(for: pros)
+            return cell
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+        // swiftlint:disable:next force_cast
+        let headerView = collectionView.dequeueReusableSupplementaryView(ofKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: HomeSectionHeaderView.identifier, for: indexPath) as! HomeSectionHeaderView
+        let editButtonTitle = "edit"
+
+        switch sections[indexPath.section] {
+        case .checkIns:
+            headerView.configure(title: sections[indexPath.section].rawValue, buttonTitle: editButtonTitle, completion: pushHabitOnboarding)
+        case .allGyms:
+            headerView.configure(title: sections[indexPath.section].rawValue, buttonTitle: editButtonTitle, completion: pushGymOnboarding)
+        case .pros:
+            headerView.configure(title: sections[indexPath.section].rawValue, buttonTitle: nil, completion: nil)
+        case .todaysClasses:
+            headerView.configure(title: sections[indexPath.section].rawValue, buttonTitle: "view all", completion: viewTodaysClasses)
```

Comment:
Could it still be worth it to move `view all` out into a `let` just to be more readable?

Reply:
i think its more readable like this actually. making it a constant will mean another variable name to look for